USING_ELEKTRA.md

```diff
@@ -0,0 +1,165 @@
+ `lcdproc`, `lcdvc` and `lcdexec`. So in the upper example we change the background
+ color of the curses driver to green for the server.
+ 
+ Elektra also comes with specifications for lcdproc such as [LCDd-spec.ini](server/specification/LCDd-spec.ini).
+ This guarantees that certain mistakes cannot occur like setting the background color to *greeen* which could
+ potentially crash the application. If you have followed the specification installation process correctly
+ from the [INSTALL.md](INSTALL.md), you now have the full specification [mounted](https://www.libelektra.org/tutorials/mount-configuration-files)
+ under the `spec` namespace. The specification for `/curses/background` background for example
+ tells us that only `red, black, green, yellow, blue, magenta, cyan, white` are valid and that the value given 
+ has to be of type string. Elektra will prevent you to set invalid values which will help reduce errors and misconfiguration.
+  All these data and much more (eg. default values, descriptions, etc.) are saved as 
+ [metadata](https://github.com/ElektraInitiative/libelektra/blob/e82b55f0d3ea4c77f3a4c04fa217084021bd8e4b/doc/dev/metadata.md)
+ for each and every key under the `spec` namespace.
+ 
+ The following sections will teach you on how to inspect all relevant settings
+  and specifications and change desired configuration settings accordingly.
+ There are multiple possibilities to do this but our focus will lie on
+ 
+ * `kdb set/get` and
+ * `kdb editor`
+ 
+ since they integrate the best with specifications for configurations.
+ 
+### Kdb set
+ 
+Using the native command line is one possibility to find out which configuration settings are available.
+If you want to see all available configuration settings for the `server`, you can simply call
+```sh
+kdb ls '/sw/lcdproc/lcdd/#0/current/server'
+#> /sw/lcdproc/lcdd/#0/current/server/autorotate
+#> /sw/lcdproc/lcdd/#0/current/server/backlight
+#> /sw/lcdproc/lcdd/#0/current/server/bind
+#> /sw/lcdproc/lcdd/#0/current/server/driver
+...
+```
+To see all available configuration settings at all you can execute `kdb ls '/sw/lcdproc/lcdd/#0/current'`.
+To see the current value you can simply call `kdb get` on the respective key.
+Now lets take the curses driver for example again and lets see what is the 
+current background color is:
+```sh
+kdb get '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> cyan
+```
+The result comes from the set `default` metadata which is saved for the key. 
+You can see all metadata by calling `kdb lsmeta`:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> check/enum
+#> check/type
+#> default
+#> description
+#> type
+```
+Here we can als see `check/enum` which is used by Elektra to assert
+for correct configuration settings. You can query for the values by calling
+`kdb getmeta`
+```sh
+kdb getmeta '/sw/lcdproc/lcdd/#0/current/curses/background' 'check/enum'
```

Comment:
You should use the array version of `check/enum`
```
check/enum = #7
check/enum/#0 = red
check/enum/#1 = black
check/enum/#2 = green
check/enum/#3 = yellow
check/enum/#4 = blue
check/enum/#5 = magenta
check/enum/#6 = cyan
check/enum/#7 = white
```

Even though this makes reading the allowed values with `kdb getmeta` harder, it is much more in line with how arrays/lists are represented in Elektra (see also ElektraInitiative/libelektra#2412)